src/Corax/Utils/EntryIdEncodings.cs

```diff
@@ -130,7 +132,7 @@ public static void DecodeAndDiscardFrequency(Span<long> entries, int read)
     {
         if (Avx2.IsSupported)
             DecodeAndDiscardFrequencyAvx2(entries, read);
-        else if (AdvSimd.IsSupported)
+        else if (IsArm == false && AdvSimd.IsSupported)
```

Comment:
Cannot we use `PlatformSpecific.IsArm`?

Reply:
Yes, in fact I would issue a different PR with a complete array of `IsXXX` apart from this PR. 
- allow us to just revert this PR when update to the underlying runtime is released
- we have a full array of switches that we can guarantee are JIT eviction compliant
What do you think?